src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -147,7 +147,28 @@ export default function Inbox ({ history: { push } }) {
   const refundAllDeclinedTransactions = useRefundAllDeclinedTransactions()
   const [counterpartyNotFound, setCounterpartyNotFound] = useState(true)
   const [isDeclinedTransactionModalVisible, setIsDeclinedTransactionModalVisible] = useState(false)
-  const [modalTransaction, setModalTransaction] = useState(null)
+  const [confirmationModalProperties, setConfirmationModalProperties] = useState({
+    shouldDisplay: false,
+    transactions: [],
+    action: '',
+    hasConfirmed: false
+  })
+
+  const resetDefaultModalTransaction = () => {
```

Comment:
I think we can call this `hideConfirmationModal` and it's easier to read and not really lying